test/integration/basic/test/hmr.js

```diff
@@ -98,6 +98,41 @@ export default (context, render) => {
 
         browser.close()
       })
+
+      // Added because of a regression in react-hot-loader, see issues: #4246 #4273
+      // Also: https://github.com/zeit/styled-jsx/issues/425
+      it('should update styles correctly', async () => {
+        const browser = await webdriver(context.appPort, '/hmr/style')
+        const pTag = await browser.elementByCss('.hmr-style-page p')
+        const initialFontSize = await pTag.getComputedCss('font-size')
+
+        expect(initialFontSize).toBe('100px')
+
+        const pagePath = join(__dirname, '../', 'pages', 'hmr', 'style.js')
+
+        const originalContent = readFileSync(pagePath, 'utf8')
+        const editedContent = originalContent.replace('100px', '200px')
+
+        // Change the page
+        writeFileSync(pagePath, editedContent, 'utf8')
+
+        // wait for 5 seconds
+        await waitFor(5000)
```

Comment:
This could be flaky or add a unnecessary delay, maybe you can make a retry function with timeout? could be done in a separate PR

Reply:
Going to merge this now, I basically copied an existing test, we could definitely improve this 👍